src/Cedar/Hub.c

```diff
@@ -5514,7 +5514,7 @@ void StorePacketToHubPa(HUB_PA *dest, SESSION *src, void *data, UINT size, PKT *
 		}
 	}
 
-	if (dest != NULL && src != NULL && dest->Session != NULL && src->Hub != NULL && src->Hub->Option != NULL)
+	if (src != NULL && dest->Session != NULL && src->Hub != NULL && src->Hub->Option != NULL)

```

Comment:
Why was dest removed from the check? dest-Session is used in the later check. There's no way for the compiler to know about the actual value of this variable at compile time.

Reply:
If `dest` is NULL, then `dest->Session` must be NULL too.

Reply:
If dest is NULL, dest-Session will cause a segmentation fault

Reply:
Right, because it dereferences a NULL pointer.

I propose to use a separated if statement to check whether `dest` is NULL.

Reply:
More specifically, it leads to undefined behavior, meaning that it might work, or it might crash. We will never know

Reply:
I created a test program to see what happens.

Segmentation fault, as expected.

Reply:
there's check already  https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/Cedar/Hub.c#L5435

if dest is NULL, we'll return from function